src/DacpacTool/Program.cs

```diff
@@ -138,6 +139,23 @@ private static int BuildDacpac(BuildOptions options)
                 packageBuilder.AddWarningsToSuppress(options.SuppressWarnings);
             }
 
+            // Add warnings suppressions for particular files through $Project.WarningsSuppression.txt
+            if (options.SuppressWarningsListFile != null)
+            {
+                if (options.SuppressWarningsListFile.Exists)
+                {
+                    foreach (var line in File.ReadLines(options.SuppressWarningsListFile.FullName))
+                    {
+                        //Checks if there are suppression warnings list
+                        var parts = line.Split('|', StringSplitOptions.RemoveEmptyEntries);
+                        var warningList = (parts.Length > 1) ? parts[1] : null;
+
+                        FileInfo inputFile = new FileInfo(parts[0]); // Validation occurs in AddInputFile
```

Comment:
The comment here doesn't match what's actually happening I think.

Reply:
OK. IMHO:
`PackageBuilder` has method `AddInputFile` - which adds the file to process at build phase, and it take one argument - `path to file`. So `PackageBuilder` has second method `AddFileWarningsToSuppress` which add suppression warnings for file. and it take two arguments - `path to file` & `suppression list`.
So `PackageBuilder` doesn't do any file parsing. It always get complete atomic argument to deal. For example in Unit Test we don't need to generate any sample `WarningsSuppression.txt`, but simple call `AddFileWarningsToSuppress` with `file` and `suppression list` arguments. `PackageBuilder` is absolutely free from knowing from what source we get arguments for his methods.
Also, IMHO, if we want to extract file parsing from `Program.cs`, we need to make middleware class between `Program` and `PackageBuilder`.

Reply:
Yes, I agree. I guess my main concern is that we don't currently test the parsing logic for the input file since we don't have any tests for Program. So yeah, this definitely needs a bit of refactoring. But I agree with @jeffrosenberg earlier comment that we should probably do that after this is merged in. So let's live this as it is for now and then we can do a follow-up-PR to refactor this.